src/testcafe-runner.js

```diff
@@ -180,18 +189,20 @@ async function run (runCfgPath, suiteName) {
 
 if (require.main === module) {
   console.log(`Sauce TestCafe Runner ${require(path.join(__dirname, '..', 'package.json')).version}`);
-  const { runCfgPath, suiteName } = getArgs();
-
-  run(runCfgPath, suiteName)
-      // eslint-disable-next-line promise/prefer-await-to-then
-      .then((passed) => {
-        process.exit(passed ? 0 : 1);
-      })
-      // eslint-disable-next-line promise/prefer-await-to-callbacks
-      .catch((err) => {
-        console.log(err);
-        process.exit(1);
-      });
+  const {runCfgPath, suiteName} = getArgs();
+  // maxTimeout maximum test execution timeout is 1800 seconds (30 mins)
+  const maxTimeout = 1800;
+
+  run(runCfgPath, suiteName, maxTimeout)
+        // eslint-disable-next-line promise/prefer-await-to-then
+        .then((passed) => {
+          process.exit(passed ? 0 : 1);
```

Comment:
If the timeout is called, `run` return `1`... which will cause process to exit as 0 (incorrectly)?

Reply:
in the case with timeout `passed` will be false and we return exit code 1.

Reply:
Yes, on timeout should exit with `1`